dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -1,59 +1,114 @@
 from datetime import datetime
-
-from dags.analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+from analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+import logging
 
 
 class DiscordTransformRawData(TransformRawDataBase):
     def transform(self, raw_data: list, platform_id: str, period: datetime) -> list[dict]:
         transformed_data = []
         for data in raw_data:
-            interactions = []
-            if data.get("replied"):
-                interactions.append({
-                    "name": "reply",
-                    "users_engaged_id": [data.get("replied")],
-                    "type": "receiver"
-                })
-
-            if data.get("user_mentions"):
-                interactions.append({
-                    "name": "mention",
-                    "users_engaged_id": data.get("user_mentions"),
-                    "type": "emitter"
-                })
-
-            if data.get("reactions"):
-                for reaction in data.get("reactions"):
-                    parts = reaction.split(", ")
-                    if len(parts) > 2:
+            try:
+                if "author" not in data or not data["author"]:
+                    raise ValueError("Missing 'author' in raw data")
+
+                interactions = []
+
+                if data.get("replied_user"):
+                    interactions.append({
+                        "name": "reply",
+                        "users_engaged_id": [data["replied_user"]],
+                        "type": "emitter"
+                    })
+                    # Create another document for the receiver
+                    receiver_interaction = {
+                        "author_id": data["replied_user"],
+                        "date": period,
```

Comment:
The date is not the `period` it would be the `createdDate` field of the raw data. Here's the data structure for raw data

https://rndadocs.notion.site/Analyzer-Discord-Pipeline-5009512d82a243deb35b3388e17b29c5#:~:text=An%20example%20for%20rawinfos%20document